File: src/test/testdata/inspections/unsafeIterators/ArrayMap.java
package main;

import com.badlogic.gdx.utils.ArrayMap;
import com.badlogic.gdx.utils.ObjectMap;

import java.util.Iterator;

class ArrayMapTest {

  ArrayMap<String, Object> map = new ArrayMap();

  void test() {

    map.put("a", new ArrayMap<String, String>().values());
    map.put("b", new ArrayMap<String, String>());

    Iterator<String> iterator1 = <warning>map.keys()</warning>;
    Iterator<Object> iterator2 = <warning>map.values()</warning>;
    Iterator<ObjectMap.Entry<String, Object>> iterator3 = <warning>map.iterator()</warning>;
    Iterator iterator4 = <warning>map.entries()</warning>;
    Iterator iterator5 = <warning>((ArrayMap)map.get("b")).values()</warning>;

    for (ObjectMap.Entry e: <warning>map</warning>) {
      System.out.println(e.key);
    }

  }

}

File: src/test/testdata/inspections/unsafeIterators/ArrayMap.kt
package main

import com.badlogic.gdx.utils.ArrayMap
import com.badlogic.gdx.utils.ObjectMap

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST")
class ArrayMapTest {

  var map: ArrayMap<String, Any> = ArrayMap()

  fun test() {

    map.put("a", <warning>ArrayMap<String, String>().values()</warning>)
    map.put("b", ArrayMap<String, String>())

    val iterator1 = <warning>map.keys()</warning>
    val iterator2 = <warning>map.values()</warning>
    val iterator3 = <warning>map.iterator()</warning>
    val iterator4 = <warning>map.entries()</warning>
    val iterator5 = <warning>(map.get("b") as ArrayMap<Any, Any>).values()</warning>

    for (e in <warning>map</warning>) {
      println(e.key)
    }

  }

}
